src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -189,6 +189,7 @@ public String getQuestionResultStatisticsHtml(
             templateToUse = FormTemplates.RANK_RESULT_RECIPIENT_STATS;
         }
         DecimalFormat df = new DecimalFormat("#.##");
+        Map<String, Integer> recipientSelfRanks = generateSelfRankForEachRecipient(responses);
```

Comment:
I believe the self rank column is only shown when the above clause is true since you only added that column for those excluding self templates, which I believe is the correct course of action as well.